update_version.sh

```diff
@@ -1,31 +1,49 @@
 #!/usr/bin/env bash
 
-if [[ ("$2" == "poetry") || (("$2" == "") && ( -f pyproject.toml)) ]]; then
+function is_output_valid() {
+    line_numbers=$(eval "echo $1 | wc -l")
+    is_version_correct=$(eval "echo $1 | grep -E '[0-9.]?'")
+
+    if [[ ($line_numbers -eq 1)  && (! -z $is_version_correct)]]; then
+        return 0
+    else
+        return 1
+    fi
+
+}
+
+if [[ ("$1" == "poetry") ]]; then
   version_file="pyproject.toml"
-  command="poetry version -s"
-elif [[ ("$2" == "version_file") || (("$2" == "") && ( -f VERSION)) ]]; then
+  command="sed 's/[\ \"]*//g' | grep -E '(^\[tool\.poetry\])|(^version)' | sed -E 's/\[tool.poetry\]//g' | sed -E 's/version[[:blank:]]?=[[:blank:]]?//g'"
+
+elif [[ ("$1" == "version_file") ]]; then
   version_file="VERSION"
-  command="cat VERSION"
+  command="grep -E '[0-9.]+'"
+
 else
-  echo "Version type '$2' not supported"
+  echo "Version type '$1' not supported"
   exit 1
 fi
 
-tfile=$(mktemp $version_file.XXXXXXXXX)
-mv $version_file $tfile
+
+version_in_branch=$(eval "cat $version_file | $command")
+
 git fetch -q origin master
-git checkout master -q -- $version_file
-version_in_master=$($command)
-mv $tfile $version_file
-version_in_branch=$($command)
+
+version_in_master=$(eval "git show origin/master:$version_file | $command")
+
+if (! is_output_valid $version_in_branch) || (! is_output_valid $version_in_master); then
+    echo "Version update is not correct"
```

Comment:
it may be beneficial to provide what is the version - in case of error it can be something else than what is in master/current branch.

Reply:
agree, can be done by moving echo commands ("Version in branch", "Version in master") before that if